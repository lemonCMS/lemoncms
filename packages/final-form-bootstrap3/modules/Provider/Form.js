import React from "react";
import PropTypes from "prop-types";
import { Form as FinalForm } from "react-final-form";
import arrayMutators from "final-form-arrays";
import ContextWrapper from "../Context/ContextWrapper";

const Form = props => {
  const { children, className, debug, readOnly, listen, ...formProps } = props;

  return (
    <FinalForm
      {...formProps}
      mutators={{
        ...arrayMutators
      }}
      render={({ handleSubmit, ...rest }) => {
        return (
          <ContextWrapper
            {...rest}
            debug={debug}
            listen={listen}
            readOnly={readOnly}
          >
            <form onSubmit={handleSubmit} className={className}>
              {children}
            </form>
          </ContextWrapper>
        );
      }}
    />
  );
};

Form.propTypes = {
  className: PropTypes.string,
  validate: PropTypes.func,
  onSubmit: PropTypes.func.isRequired,
  children: PropTypes.oneOfType([PropTypes.object, PropTypes.array]).isRequired,
  debug: PropTypes.bool,
  readOnly: PropTypes.bool,
  listen: PropTypes.func,
  initialValues: PropTypes.objectOf(
    PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.number,
      PropTypes.array,
      PropTypes.object,
      PropTypes.bool
    ])
  )
};

Form.defaultProps = {
  className: "",
  validate: () => {},
  listen: () => {},
  debug: false,
  readOnly: false,
  initialValues: {}
};

export default Form;
