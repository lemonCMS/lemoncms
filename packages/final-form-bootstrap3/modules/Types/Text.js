import React from "react";
import PropTypes from "prop-types";
import { FormControl } from "react-bootstrap";
import context from "../decorators/context";
import fieldGroup from "./fieldGroup";

const Text = props => {
  const { input, type, isDisabled } = props;
  return <FormControl {...input} type={type} disabled={isDisabled} />;
};

Text.propTypes = {
  input: PropTypes.shape({
    name: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
    onBlur: PropTypes.func.isRequired,
    onFocus: PropTypes.func.isRequired,
    value: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.bool,
      PropTypes.number
    ])
  }),
  placeholder: PropTypes.string,
  label: PropTypes.string,
  help: PropTypes.string,
  computedInvalid: PropTypes.bool.isRequired,
  type: PropTypes.oneOf(["text", "email", "date", "datetime-local", "checkbox"])
};

Text.defaultProps = {
  input: {},
  label: null,
  help: null,
  placeholder: null,
  type: "text"
};

export default context()(fieldGroup(Text));
