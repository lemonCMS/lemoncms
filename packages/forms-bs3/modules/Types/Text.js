import React from "react";
import PropTypes from "prop-types";
import { FormControl } from "react-bootstrap-v3";
import context from "../decorators/context";
import fieldGroup from "./fieldGroup";

const Text = props => {
  const { input, type } = props;
  return <FormControl {...input} type={type} />;
};

Text.propTypes = {
  input: PropTypes.oneOfType([PropTypes.object]),
  placeholder: PropTypes.string,
  label: PropTypes.string,
  help: PropTypes.string,
  computedInvalid: PropTypes.bool.isRequired,
  type: PropTypes.oneOf(["text", "email", "date", "datetime-local", "checkbox"])
};

Text.defaultProps = {
  input: {},
  label: null,
  help: null,
  placeholder: null,
  type: "text"
};

export default context(fieldGroup(Text));
