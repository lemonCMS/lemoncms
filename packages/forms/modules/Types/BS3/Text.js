import React from "react";
import PropTypes from "prop-types";
import {
  FormGroup,
  ControlLabel,
  FormControl,
  HelpBlock
} from "react-bootstrap-v3";
import context from "../../decorators/context";
import field from "../../decorators/field";

const Text = props => {
  const { input, placeholder, computedInvalid, label, help } = props;

  return (
    <FormGroup validationState={computedInvalid ? "error" : null}>
      {label && <ControlLabel>{label}</ControlLabel>}
      <FormControl type="text" placeholder={placeholder} {...input} />
      <FormControl.Feedback />
      {help && <HelpBlock>{help}</HelpBlock>}
    </FormGroup>
  );
};

Text.propTypes = {
  input: PropTypes.oneOfType([PropTypes.object]),
  placeholder: PropTypes.string,
  label: PropTypes.string,
  help: PropTypes.string,
  computedInvalid: PropTypes.bool.isRequired
};

Text.defaultProps = {
  input: {},
  label: null,
  help: null,
  placeholder: null
};

export default context(field(Text));
