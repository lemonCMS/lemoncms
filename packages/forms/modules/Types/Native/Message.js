import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import consumer from '../../decorators/consumer';

const Message = props => {
  const {
    type,
    className,
    context: {
      status: {
        invalid, valid, submitSucceeded, submitFailed
      }
    },
    children
  } = props;

  if (type === 'success') {
    if (valid && submitSucceeded) {
      return <div className={classNames(['message-success', className])}>{children}</div>;
    }
    return null;
  }

  if (invalid && submitFailed) {
    return <div className={classNames(['message-error', className])}>{children}</div>;
  }

  return null;
};

Message.propTypes = {
  type: PropTypes.string.isRequired,
  className: PropTypes.string,
  context: PropTypes.shape({
    status: PropTypes.shape({
      invalid: PropTypes.bool,
      valid: PropTypes.bool,
      submitFailed: PropTypes.bool,
      submitSucceeded: PropTypes.bool
    })
  }),
  children: PropTypes.oneOfType([PropTypes.element, PropTypes.string]).isRequired
};

Message.defaultProps = {
  className: '',
  context: {}
};

export default consumer(Message);
