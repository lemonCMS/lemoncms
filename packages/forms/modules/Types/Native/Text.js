import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import context from '../../decorators/context';
import field from '../../decorators/field';

const Text = props => {
  const {
    input, placeholder, formControl, computedInvalid
  } = props;
  return (
    <input
      type="text"
      placeholder={placeholder}
      className={classNames('ffc-input', { 'is-invalid': computedInvalid }, formControl)}
      {...input}
    />
  );
};

Text.propTypes = {
  input: PropTypes.oneOfType([PropTypes.object]),
  placeholder: PropTypes.string,
  formControl: PropTypes.string,
  computedInvalid: PropTypes.bool.isRequired
};

Text.defaultProps = {
  input: {},
  placeholder: null,
  formControl: null
};

export default context(field(Text));
