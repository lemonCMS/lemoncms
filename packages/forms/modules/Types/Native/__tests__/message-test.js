import React from 'react';
import ReactDOM from 'react-dom';
// import { act } from 'react-dom/test-utils';
import renderer, {act} from 'react-test-renderer';
import Form from '../../../Provider/Form';
import Message from '../Message';
import Text from '../Text';

let container;

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  document.body.removeChild(container);
  container = null;
});

describe('Message', () => {

  it('should display success a message', () => {
    const submit = jest.fn();
    submit.mockReturnValueOnce({});
    submit.mockReturnValueOnce({username: 'invalid'});

    const Component = () => <Form onSubmit={submit} initialValues={{ username: 'raymond' }}>
      <Text type={'text'} name={'username'} />
      <Message type={'success'}>This success message should show</Message>
      <Message type={'error'}>This error message should show</Message>
      <button type={'submit'}>do it</button>
    </Form>;

    act(() => {
      ReactDOM.render(<Component />, container);
    });

    const button = container.querySelector('button');

    // Run success message
    act(() => {
      button.dispatchEvent(new MouseEvent('click', { bubbles: true }));
    });

    expect(submit).toHaveBeenCalledTimes(1);
    expect(submit).toHaveBeenCalledWith({username: 'raymond'}, expect.any(Object), expect.any(Function));
    expect(container.querySelector('div[class="message-success"]')).toBeTruthy();
    expect(container.querySelector('div[class="message-error"]')).toBeFalsy();

    // Run error message
    act(() => {
      button.dispatchEvent(new MouseEvent('click', { bubbles: true }));
    });

    expect(submit).toHaveBeenCalledTimes(2);
    expect(submit).toHaveBeenCalledWith({username: 'raymond'}, expect.any(Object), expect.any(Function));
    expect(container.querySelector('div[class="message-success"]')).toBeFalsy();
    expect(container.querySelector('div[class="message-error"]')).toBeTruthy();
  });
});
