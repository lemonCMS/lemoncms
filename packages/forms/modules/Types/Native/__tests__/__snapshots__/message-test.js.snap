// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message should display success a message 1`] = `
<form
  className=""
  onSubmit={[Function]}
/>
`;
